src/components/editor/SinopiaResourceTemplates.jsx

```diff
@@ -23,63 +23,69 @@ class SinopiaResourceTemplates extends Component {
   }
 
   async componentDidUpdate(prevProps) {
-    if (this.props.updateKey > prevProps.updateKey) {
-      const groupPromise = getGroups().then(data => {
-        return data
-      }).catch(() => {})
+    if (this.props.updateKey <= prevProps.updateKey)
+      return
 
-      await this.fulfillGroupPromise(groupPromise).then(async () => {
-        this.state.groupData.map(group => {
+    getGroups()
+      .then(data => {
+        const contains = [].concat(data.response.body.contains)
+        this.setState({groupData: contains})
+      })
+      .catch(error => {
+        this.setState({message: error})
+      })
+      .then(() => {
+        return this.state.groupData.map(group => {
           const groupName = this.resourceToName(group)
-          listResourcesInGroupContainer(groupName).then((data) => {
-            this.fulfillGroupData(data)
-          }).catch(() => {})
+          return listResourcesInGroupContainer(groupName)
+        })
+      })
+      .then(groupResources => {
+        Promise.all(groupResources).then(responses => {
+          responses.forEach(group => {
+            this.handleGroupResponse(group)
+          })
         })
       })
-    }
   }
 
-  fulfillGroupPromise = (promise) => {
-    promise.then(data => {
-      const contains = [].concat(data.response.body.contains)
-      this.setState({groupData: contains})
-    }).catch((error) => {
-      this.setState({message: error})
-    })
-    return promise
-  }
+  handleGroupResponse = data => {
```

Comment:
consider renaming to `setStatePerTrellis` or whatever. ("handle" is often "handleClick" sort of thing)

Reply:
done in my local branch